openedx/adg/lms/webinars/tests/test_views.py

```diff
@@ -0,0 +1,81 @@
+"""
+All test cases for webinars app views
+"""
+import pytest
+from django.urls import reverse
+
+from common.djangoapps.student.tests.factories import UserFactory
+from openedx.adg.lms.webinars.models import Webinar, WebinarRegistration
+from openedx.adg.lms.webinars.tests.factories import WebinarFactory, WebinarRegistrationFactory
+
+pytestmark = pytest.mark.django_db
+
+
+@pytest.fixture(name='user_client')
+def user_client_login(request, client):
+    """
+    User and client login fixture. User will be authenticated for all tests where we pass this fixture.
+    """
+    user = UserFactory()
+    client.login(username=user.username, password='test')
+    return user, client
+
+
+def test_webinar_registration_view_object_does_not_exist(user_client):
+    """
+    Test webinar registration if webinar does not exist
+    """
+    _, client = user_client
+
+    response = client.post(reverse('webinar_registration', kwargs={'pk': 100, 'action': 'register'}))
+
+    assert response.status_code == 404
+
+
+def test_webinar_registration_view_cancelled_webinar(client, user_client):
+    """
+    Test webinar registration if webinar is cancelled
+    """
+    _, client = user_client
+    webinar = WebinarFactory(status=Webinar.CANCELLED)
+
+    response = client.post(reverse('webinar_registration', kwargs={'pk': webinar.id, 'action': 'register'}))
+
+    assert response.status_code == 500
+
+
+@pytest.mark.parametrize('action', ['register', 'cancel'])
+def test_webinar_registration_view_register_user_with_no_prior_registration(action, user_client, mocker):
+    """
+    Test webinar registration and cancellation if webinar was not previously registered by user
+    """
+    mock_send_registration_email = mocker.patch('openedx.adg.lms.webinars.views.send_webinar_registration_email')
+    webinar = WebinarFactory(status=Webinar.UPCOMING)
+    user, client = user_client
+
+    client.post(reverse('webinar_registration', kwargs={'pk': webinar.id, 'action': action}))
+
+    if action == 'register':
+        mock_send_registration_email.assert_called_once_with(webinar, user.email)
+    else:
+        mock_send_registration_email.assert_not_called()
+
+
+@pytest.mark.parametrize('action', ['register', 'cancel'])
+@pytest.mark.parametrize('is_registered', [True, False])
+def test_webinar_registration_view_register_user_with_prior_registration(action, user_client, is_registered, mocker):
+    """
+    Test webinar registration and cancellation if webinar was previously registered or canceled by user
+    """
+    mock_send_registration_email = mocker.patch('openedx.adg.lms.webinars.views.send_webinar_registration_email')
+    user, client = user_client
+    registration = WebinarRegistrationFactory(user=user, is_registered=is_registered)
+
+    client.post(reverse('webinar_registration', kwargs={'pk': registration.webinar.id, 'action': action}))
```

Comment:
Where can we reverse `webinar_registration` from?

Reply:
I have updated code, please re-review